src/main/scala/xiangshan/frontend/icache/WayLookup.scala

```diff
@@ -132,9 +158,13 @@ class WayLookup(implicit p: Parameters) extends ICacheModule {
     * read
     ******************************************************************************
     */
-  val bypass = empty && io.write.valid
-  io.read.valid             := !empty || io.write.valid
-  io.read.bits              := Mux(bypass, io.write.bits, entries(readPtr.value))
+  io.read.valid := !empty || io.write.valid
+  when (empty && io.write.valid) {  // bypass
+    io.read.bits := io.write.bits
+  }.otherwise {
+    io.read.bits.entry := entries(readPtr.value)
+    io.read.bits.gpf   := Mux(readPtr === gpfPtr, gpf_entry, 0.U.asTypeOf(new WayLookupGPFEntry))
```

Comment:
readPtr === gpfPtr may be a bug. I encountered a bug that gpfPtr became zero and readPtr became zero after readPtr increased to 3f which make readPtr === gpfPtr. An instruction happens igpf, but that shouldn't happen.

Reply:
This is expected, every detected gpf will cause a flush (either redirect due to misprediction, or exception handling), and the gpf_entry will be reset at the time of the flush, ensuring that subsequent requests will not read gpf even if `readPtr === gpfPtr`.

It should be noted that `readPtr === gpfPtr` may result in reading out the wrong gpf **after the first gpf occurs and before the flush**, but at that point it's already on the wrong execution path, so it actually doesn't matter.

The root cause of the bug is #3215.